platform/ios/test/MGLMapViewLayoutTests.m

```diff
@@ -0,0 +1,119 @@
+
+@end
+
+@implementation MGLMapViewLayoutTests
+
+- (void)setUp {
+    [super setUp];
+
+    [MGLAccountManager setAccessToken:@"pk.feedcafedeadbeefbadebede"];
+    NSURL *styleURL = [[NSBundle bundleForClass:[self class]] URLForResource:@"one-liner" withExtension:@"json"];
+    NSLog(@"================> Screen size: %@", NSStringFromCGRect(UIScreen.mainScreen.bounds));
+
+    self.superView = [[UIView alloc] initWithFrame:UIScreen.mainScreen.bounds];
+
+    self.mapView = [[MGLMapView alloc] initWithFrame:UIScreen.mainScreen.bounds styleURL:styleURL];
+    self.mapView.delegate = self;
+
+    [self.superView addSubview:self.mapView];
+
+    UIView *mapView = self.mapView;
+    NSDictionary *bindings = NSDictionaryOfVariableBindings(mapView);
+    NSArray *verticalConstraints = [NSLayoutConstraint constraintsWithVisualFormat:@"V:|-0-[mapView]-0-|" options:0 metrics:nil views:bindings];
+    NSArray *horizonatalConstraints = [NSLayoutConstraint constraintsWithVisualFormat:@"H:|-0-[mapView]-0-|" options:0 metrics:nil views:bindings];
+
+    [self.superView addConstraints:[verticalConstraints arrayByAddingObjectsFromArray:horizonatalConstraints]];
+
+    self.styleLoadingExpectation = [self expectationWithDescription:@"Map view should finish loading style."];
+    [self waitForExpectationsWithTimeout:1 handler:nil];
+
+    //set zoom and heading so that scale bar and compass will be shown
+    [self.mapView setZoomLevel:4.5 animated:NO];
+    [self.mapView.camera setHeading:12.0];
+
+    //invoke layout
+    [self.superView setNeedsLayout];
+    [self.superView layoutIfNeeded];
+    NSLog(@"================> self.mapView.frame: %@", NSStringFromCGRect(self.mapView.frame));
+    if ( [self.mapView respondsToSelector:@selector(safeAreaInsets)] ) {
+        NSLog(@"================> self.mapView.safeAreaInsets: %@", NSStringFromUIEdgeInsets(self.mapView.safeAreaInsets));
+    }
+}
+
+- (void)mapView:(MGLMapView *)mapView didFinishLoadingStyle:(MGLStyle *)style {
+    XCTAssertNotNil(mapView.style);
+    XCTAssertEqual(mapView.style, style);
+
+    [self.styleLoadingExpectation fulfill];
+}
+
+- (void)tearDown {
+    self.styleLoadingExpectation = nil;
+    self.mapView = nil;
+
+    [super tearDown];
+}
+
+// TODO: check constraining to superview vs. constraining to safeAreaLayoutGuide.bottomAnchor
+
+- (void)testOrnamentPlacement {
+
+    CGFloat margin = 8.0;
+    CGFloat bottomSafeAreaInset = 0.0;
+    double accuracy = 0.01;
+
+    if ( [self.mapView respondsToSelector:@selector(safeAreaInsets)] ) {
+        bottomSafeAreaInset = self.mapView.safeAreaInsets.bottom;
+    }
+    
+    //compass
+    UIImageView *compassView = self.mapView.compassView;
+    NSLog(@"================> %@", compassView);
+
+    CGFloat expectedCompassOriginX = CGRectGetMaxX(self.mapView.bounds) - margin - CGRectGetWidth(compassView.frame);
+    CGFloat expectedCompassOriginY = margin;
+    // what about width/height? maybe we don't care as much at the moment?
```

Comment:
Let's either inspect width/height or remove the comment. This was just here as a placeholder while we were defining the test case.